app/scripts/directives/loading.dir.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+angular.module('yourStlCourts').directive('loading', function($timeout){
+  return{
+    restrict: 'E',
+    scope:{
+    },
+    templateUrl: 'views/loading.html',
+    link: function($scope,element,attrs){
+      var fastEnough = true;
+      $(element).addClass("hidden");
+      $scope.$on('$stateChangeStart',function(){
+        fastEnough = false;
+        $timeout(function(){
+          if (!fastEnough){
+            $(element).removeClass("hidden");
+          }
+        },500);
```

Comment:
This actually suffers from one of the more annoying UI/UX problems, of flickering screens. Imagine a thing that takes 550 ms to load. for half a second, nothing will happen, and then barely noticeably you will see something flash on screen and then disappear.

A good short term screen element is on screen when it becomes relevant (like when a state is changing) and on screen for a minimum amount of time (usually half a second or more) even if that takes more time than the thing it indicated.